.github/workflows/make-release.yml

```diff
@@ -24,35 +24,47 @@ jobs:
       - name: Checkout code
         uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11  # v4.1.1
         with:
-          # Here `token` is needed to avoid incurring in error GH006 Protected  Branch Update Failed,
-          token: ${{ secrets.GH_PUBLISH_TOKEN }}
-          # While `fetch-depth` is used to allow the workflow to later commit & push the changes.
-          fetch-depth: 0
+          ref: ${{ github.sha }}
       - name: Setup NodeJS
         uses: actions/setup-node@b39b52d1213e96004bfcb1c61a8a6fa8ab84f3e8 # v4.0.1
         with:
           node-version: "20"
           cache: "npm"
       - name: Setup auth tokens
         run: |
-          git config --global user.name 'github-actions[bot]'
-          git config --global user.email 'github-actions[bot]@users.noreply.github.com'
-          git remote set-url origin https://x-access-token:${{ secrets.GH_PUBLISH_TOKEN }}@github.com/$GITHUB_REPOSITORY
           npm set "//registry.npmjs.org/:_authToken=${{ secrets.NPM_TOKEN }}"
       - name: Setup dependencies
         uses: ./.github/actions/cached-node-modules
-      - name: Version
-        run: |
-          npx lerna version minor --force-publish --no-commit-hooks --yes
       - name: Publish to npm
         run: |
-          NPM_CONFIG_PROVENANCE=true npx lerna publish from-git --yes
+          NPM_CONFIG_PROVENANCE=true npx lerna publish from-package --git-head ${{ github.sha }} --yes 
```

Comment:
NIIIIIIICE! Such a tiny change with huge value <3 thank you

Reply:
Can you please test this locally against [`verdaccio`](https://verdaccio.org) to see how it behaves when re-run? 

Since we publish multiple packages there could be cases in which some packages are published successfully and others aren't and we need to repeat the run.

Basically I'd like to make sure the release workflow is idempotent. At least based on the docs for these flags this seems the most appropriate command but I'd like to err on the caution side and test it, if possible.

Reply:
I used both, `verdaccio` locally and private playground repo to publish, it worked and it is also idempotent. The command will compare the package version with the registry and publish only the package versions that are higher. Here is an example with two packages, where one is already publish on NPM and the other is not:

```bash
Found 1 package to publish:
lerna info auto-confirmed 
lerna info publish Publishing packages to npm...
 - @am29d/commons => 1.0.13
```

This raises a more general question on how we should see the release operation in case of partial failure. If it's atomic, we'd need to investigate the issue and make another release of all packages. Or, we'd need to treat each package as an individual release with the convention to have the same version across all packages (which is an indirect dependency). 

TLDR: rerunning the job will publish only unpublished packages with the higher local version compared to NPM registry.

Reply:
> This raises a more general question on how we should see the release operation in case of partial failure.

If the failure is in the CI pipeline (for whatever reason), then a retry, even if it involves commits to fix our workflows, should not warrant a version bump imo. 

I see a lot of projects that bump the version whenever there's a change in dev dependencies (on the project's side) or in their pipelines, and imo this is wrong. Version bumps should happen only when the code published to npm changes.

With this in mind, retrying should be ok, as long as the goal is to have all the packages of a given version published to npm.

> TLDR: rerunning the job will publish only unpublished packages with the higher local version compared to NPM registry.

I think this is technically not accurate, and if it was, it would prevent us from ever publishing a v1 security release once we release v2. After reading the wording on the `lerna version` docs my understanding is that lerna simply reads the version in th `package.json`, then checks if that version is available on the registry, and if not it publishes it.

Which one is it?

Reply:
Also, sorry for not mentioning this in my previous message, but thanks a lot for taking the time to test this. Amazing work!

Reply:
I might be wrong, but as far as I understood, `lerna version` operates on the repo, not on the NPM. Did you mean `lerna publish`? Then this is correct, we can not publish v1 fixes using `from-package` once v2 is published in NPM, **using this workflow**. 

We can't have both, using `from-package` helps us to rerun the workflow, using `from-git` would publish any version we have in the `package.json`. 

If the intention is to have one workflow for both releases, then I need to revisit the workflow.

Reply:
Sorry, it was a typo. I meant `publish`, and I was referring to the sentence: `rerunning the job will publish only unpublished packages with the higher local version compared to NPM registry.`.

If this was true then once we have say `2.0.0` on npm, then attempting to publish `1.20.0` would not succeed because the local version is lower than the one on the registry.

But this is not the case, the `lerna` docs say this:
> the list of packages to publish is determined by inspecting each package.json and determining if any package version is not present in the registry. Any versions not present in the registry will be published.

So we are good